sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1165,7 +1165,7 @@ class Analyzer(
  * scoping information for attributes and can be removed once analysis is complete.
  */
 object EliminateSubQueries extends Rule[LogicalPlan] {
-  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
```

Comment:
for future reference, `transformDown` doesn't work because of: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala#L267

After apple rule to current node, it goes to its children next. So for a nested subquery, like `Sub(Sub(abc))`, it first apply the rule and got `Sub(abc)`, then goes to children, i.e. `abc`, so the mid subquery is skipped.